ts/src/time-profiler-inspector.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {TimeProfile, TimeProfileNode} from './v8-types';
+import * as inspector from 'node:inspector';
+
+const session = new inspector.Session();
+session.connect();
+
+// Wrappers around inspector functions
+export function startProfiling(): Promise<void> {
+  return new Promise<void>((resolve, reject) => {
+    session.post('Profiler.enable', err => {
+      if (err !== null) {
+        reject(err);
+      }
+      session.post('Profiler.start', err => {
+        if (err !== null) {
+          reject(err);
+        }
+        resolve();
+      });
+    });
+  });
+}
+
+export function stopProfiling(): Promise<TimeProfile> {
+  // return profiler.timeProfiler.stopProfiling(runName, includeLineInfo || false);
+  return new Promise<TimeProfile>((resolve, reject) => {
+    session.post('Profiler.stop', (err, {profile}) => {
+      if (err !== null) {
+        reject(err);
+      }
+      resolve(translateToTimeProfile(profile));
+    });
+  });
+}
+
+function translateToTimeProfile(
+  profile: inspector.Profiler.Profile
+): TimeProfile {
+  const root: inspector.Profiler.ProfileNode | undefined = profile.nodes[0];
+  // Not sure if this could ever happen...
+  if (root === undefined) {
+    return {
+      endTime: profile.endTime,
+      startTime: profile.startTime,
+      topDownRoot: {
+        children: [],
+        hitCount: 0,
+        scriptName: '',
+      },
+    };
+  }
+
+  const nodesById: {[key: number]: inspector.Profiler.ProfileNode} = {};
+  profile.nodes.forEach(node => (nodesById[node.id] = node));
+
+  function translateNode({
+    hitCount,
+    children,
+    callFrame: {columnNumber, functionName, lineNumber, scriptId, url},
+  }: inspector.Profiler.ProfileNode): TimeProfileNode {
+    const parsedScriptId = parseInt(scriptId);
+    return {
+      name: functionName,
+      scriptName: url,
```

Comment:
Url will have a `file:/` scheme, which differs from the C++ impl

Reply:
This is probably a breaking change from the perspective of profile consumers (Cloud Profiler's UI doesn't have a way to ignore file names when doing profile comparison -- so the flame graph will fork when merging profiles across versions & doing diff). But, other than that, I don't see this posing any major issues.

Reply:
Added code to strip this prefix when present anyway. I imagine the URI is mainly intended for browser scripts but for node will usually be `file:/` and repetitive.